src/EventAccumulator.ts

```diff
@@ -81,19 +118,28 @@ export class EventAccumulator implements IEventAccumulator {
    * @param {string} key - event key
    */
   flush (key: string): void {
-    // if event is not in the store, return.
-    if (!(key in this.events)) {
+    const eventStore = this.events.get(key)
+    if (eventStore == null) {
       return
     }
 
-    const { eventData, startTime, timeout } = this.events[key]
+    const { eventData, startTime, timeout } = eventStore
 
     // update duration to ms from start.
     eventData.dur = Date.now() - startTime
```

Comment:
I think we should remove duration from the submitted event though, because countly doesn't really display them the way we think they should.. maybe we should chat about this while taking a peek at existing webui data in our countly server